src/topicmodeller/topicmodeller/topicmodeller.py

```diff
@@ -0,0 +1,155 @@
+            # To avoid poor performances on __filter_document method, dictionary words
+            # are "cached" on a set (O(1)).
+            self.dictionary_words = set(self.dictionary.values())
+
+        if os.path.isfile(self.lda_model_file):
+            self.lda = models.LdaModel.load(self.lda_model_file)
+
+    def initialize_dictionary(self, batched_documents):
+        self.dictionary = corpora.Dictionary()
+
+        for (_, documents) in batched_documents:
+            corpora.Dictionary.add_documents(self.dictionary, documents)
+        self.dictionary.save(self.dictionary_file)
+
+        self.dictionary_words = set(self.dictionary.values())
+
+        self.lda = None
+
+    def feed(self, batched_documents):
+        for (_, documents) in batched_documents:
+            # Construct the Corpus : Transform each document to a vector [(word_id, word_count) | word_count > 0]
+            corpus = [self.dictionary.doc2bow(self.__filter_document(document)) for document in documents]
+            self.__update_model(corpus)
+        self.lda.save(self.lda_model_file)
+
+    def classify(self, document):
+        return self.lda[self.dictionary.doc2bow(document)]
+
+    def __update_model(self, new_corpus):
+        if self.lda is None:
+            self.lda = models.LdaModel(new_corpus, id2word=self.dictionary, num_topics=self.topics)
+        else:
+            self.lda.update(new_corpus)
+
+    def __filter_document(self, document):
+        # We want to keep only known words (those on our dictionary)
+        return [word for word in document if word in self.dictionary_words]
+
+
+# From JSON to scraper.Document object
+def _decode(content):
+    return jsonpickle.decode(content)
+
+
+# Extract a readable document from HTML
+def _readable_document(html_document):
+    return Extractor(extractor=u'ArticleExtractor', html=html_document).getText()
+
+
+def _word_tokenize(content):
+    word_tokenized = []
+    for sentence in nltk.sent_tokenize(content):
+        word_tokenized += nltk.word_tokenize(sentence)
+    return word_tokenized
+
+
+def _clean(words):
+    return _filter_latin_words(_remove_stop_words(words))
+
+
+def _remove_stop_words(words):
+    return [word for word in words if word not in stopwords.words('english')]
+
+
+def _filter_latin_words(words):
+    return [word for word in words if re.search(r'^[a-zA-Z\-\']*$', word) is not None]
+
+
+def initialize_model(documents_folder, tm_data_folder):
+    documents = RepeatableBatchedDocuments(documents_folder, 2000)
+
+    topicmodeller = TopicModeller(tm_data_folder, 128)
+
+    topicmodeller.initialize_dictionary(documents)
+    topicmodeller.feed(documents)
+
+
+def classify_and_dump_json(documents_folder, tm_data_folder, output_folder):
+    topicmodeller = TopicModeller(tm_data_folder, 128)
+
+    jsonpickle.set_encoder_options('simplejson', indent=4, ensure_ascii=False)
+
+    documents = RepeatableBatchedDocuments(documents_folder, 2000)
+
```

Comment:
cf. our discussion, i think
TopicModeller(tm_data_folder, nb_topics=128) or RepeatableBatchedDocuments(documents_folder, batch_size=2000)
is clearer, or I ask myself "what is the 128" the first time i see it, and I don't have resharper :-) .